### ISSUE TITLE
Enable configuration of podRetention

### ISSUE THREAD

**Issue by oalyman** [opendevstack/ods-jenkins-shared-library#125]
If I understand the [current implementation](https://github.com/opendevstack/ods-jenkins-shared-library/blob/6beb90bcb1c61de36a7f7c18a3f4503e7c6bd743/src/org/ods/OdsPipeline.groovy#L59) correctly there is no way to configure [Agent Pod Retention](https://docs.openshift.com/container-platform/3.11/using_images/other_images/jenkins_slaves.html#agent-pod-retention).

It would be of great benefit to be able to to do so.

**Comment by michaelsauter** [comment]
Yes, that would be great. Keep in mind that this only "works" if the label is set to a static value as well ...

**Comment by michaelsauter** [comment]
As this isn't crucial, moving to v4.
